_posts/2020-02-14-diy-drones-digital.md

```diff
@@ -0,0 +1,171 @@
+        - If not, then depending upon the RPM sends a specific error code.
+
+- **In-Flight Motor Status**
+    - The ESC can report back status like:
+        - Actual RPM
+        - Current Draw
+        - Motor Temps
+    - These in flight status will allow the main controller to decide if it can over-drive the motors or if it should limit the stress and under-drive them, and do this dynamically in real time.
+    - In case of active ESC and Motor cooling, the main controller will be able to regulate that as well.
+    - This will further help set different flight profile, similar to that of a modern day sports car:
+        - Long Range mode will under-drive the motors so that they don't waste too much energy on heat.
+        - Balanced mode: as it says a nice balance between range and performance.
+        - Race mode: Pushes the motors to their absolute limits and disregards flight time.
+        - And everything in between.
+
+- **Motor spin up curve:**
+    - Using the digital interface between our main controller and the ESC we can create a motor spin-up map.
+    - We can program it for:
+        - Soft takeoffs and landing
+        - Hard takeoff and soft landing
+        - Damp or hard acceleration, etc.
+
+- Some other things like:
+    - **Firmware-less ESC:** where the ESC firmware is loaded by the main controller during boot.
+    - **Encrypted Transceivers:** where the commutation channel between the hand held controller and the drone can be encrypted.  
+
+***
+
+## Difficulties:
+
+- **Cost:** Everything from the transmitter to the on-board receiver will need to be re-engineered. The cost is not the matter of BOM, MCUs are extremely cheap. Its a matter of introducing an entirely new methodology into a well established ecosystem, thereby being at a risk of selling in lower quantities.
+The all digital drone dream would need to be very price competitive from the beginning.
+
+- **Video Transmission:** All FPV drones today use analogue video as it offers near-zero latency, albeit at craptastic quality. This is something that requires serious engineering and dough to achieve. It will not only help in streaming high quality footage but also help with in-flight visual inference.
+
+***
+***
+
+# In Conclusion
+
+The Idea of a fully digital interface drone came about when I was building out my first drone, I was looking at the ESCs and thought to myself that a stm32 part on something as simple as a ESC is probably way too overkill and since then I was always under the impression that the ESC was able to do a lot more. And of course I was disappointed, when I finally started building the drone and was face to face with the limited functionality of the ESC.
+
+Since then, I've had this idea of a Drone ESC Mezzanine that would provide a much smarter ESC which would in-turn enable us too make a fully automated and pilot-less drones for close the the price of a FPV Drone.
```

Comment:
```suggestion
Since then, I've had this idea of a Drone ESC Mezzanine that would provide a much smarter ESC which would in-turn enable us too make a fully automated and pilot-less drones for close the price of a FPV Drone.
```